localstack/aws/handlers/response_aggregator.py

```diff
@@ -0,0 +1,64 @@
+import atexit
+import dataclasses
+import datetime
+import json
+import logging
+import threading
+from typing import Any, Dict
+
+from localstack.aws.api import RequestContext
+from localstack.aws.chain import HandlerChain
+from localstack.http import Response
+from localstack.utils import analytics
+from localstack.utils.scheduler import Scheduler
+
+LOG = logging.getLogger(__name__)
+FLUSH_INTERVAL_SECS = 10
+
+
+@dataclasses.dataclass
+class ResponseInfo:
+    service: str
+    operation: str
+    status_code: int
+
+    def to_string(self):
+        return json.dumps(dataclasses.asdict(self), sort_keys=True)
+
+
+class ResponseAggregator:
+    def __init__(self):
+        self.response_counts = {}
+        self.period_start_time = datetime.datetime.utcnow()
+        self.flush_scheduler = Scheduler()
+        self.scheduler_thread = threading.Thread(target=self.flush_scheduler.run)
+        self.scheduler_thread.start()
+        self.flush_scheduler.schedule(func=self.flush, period=FLUSH_INTERVAL_SECS, fixed_rate=True)
+        atexit.register(self.flush)
```

Comment:
it would be great if we could move most of the code here into the `localstack.utils.analytics` package, and just leave the handler with a simple call to some abstraction that sits in the analytics package. so separate `ResponseAggregator` from the `Handler` implementation.